scancodeio/wsgi.py

```diff
@@ -33,6 +33,27 @@
 
 from django.core.wsgi import get_wsgi_application
 
+from celery.exceptions import OperationalError
+
+from scancodeio.celery import app
+from scanpipe.models import Run
+
 os.environ.setdefault("DJANGO_SETTINGS_MODULE", "scancodeio.settings")
 
+
+def get_celery_worker_status():
+    try:
+        i = app.control.inspect()
+        active = i.active()
+        scheduled = i.scheduled()
+        if active[list(active.keys())[0]] or scheduled[list(scheduled.keys())[0]]:
+            return True
+    except OperationalError:
+        return False
+
+
+result = get_celery_worker_status()
+if not result:
+    Run.objects.filter(task_exitcode__isnull=True).update(task_exitcode=1)
```

Comment:
Can you elaborate what this does?

Reply:
`Run.objects.filter(task_exitcode__isnull=True).update(task_exitcode=1)` - This command is executed when no worker is available at the moment when server is started. The filter extracts all entries having `task_exitcode = NULL`, which represents `Running` state,  and updates them to represent failure in execution `task_exitcode = 1`.